src/lib.rs

```diff
@@ -48,5 +53,73 @@
 #![cfg_attr(feature="clippy", deny(clippy, unicode_not_nfc, wrong_pub_self_convention,
                                    option_unwrap_used))]
 
-#[test]
-fn place_holder() {}
+
+#[cfg(target_os = "macos")]
+extern crate libc;
+
+mod app;
+pub use app::App;
+
+
+
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "linux")]
+pub use linux::{install, open};
+
+
+
+#[cfg(target_os = "windows")]
+mod windows;
+#[cfg(target_os = "windows")]
+pub use windows::{install, open};
+
+
+
+#[cfg(target_os = "macos")]
+mod macos;
+#[cfg(target_os = "macos")]
+pub use macos::{install, open};
+
+
+/// Foreign Function call Interface to use this library
+pub mod ffi {
+	#![allow(unsafe_code, unused)]
+
+	use std::ffi::CString;
+	use std::os::raw::c_char;
+	use super::{App, open as rust_open, install as rust_install};
+
+	#[no_mangle]
+	/// open the given URI on this system
+	pub unsafe extern "C" fn open(uri: *mut c_char) -> bool {
```

Comment:
What is the benefit of that over using `c_char`? All examples (outside our code) I found in the rust world seem to be using c_char.